realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/AuthTests.java

```diff
@@ -144,6 +140,84 @@ public void onError(ObjectServerError error) {
         });
     }
 
+    @Test
+    @RunTestInLooperThread
+    public void login_withAnonymous() {
+        SyncCredentials credentials = SyncCredentials.anonymous();
+        SyncUser.loginAsync(credentials, Constants.AUTH_URL, new SyncUser.Callback<SyncUser>() {
+            @Override
+            public void onSuccess(SyncUser user) {
+                assertFalse(user.isAdmin());
+                final SyncConfiguration config = new SyncConfiguration.Builder(user, Constants.SYNC_SERVER_URL)
+                        .errorHandler((session, error) -> fail("Session failed: " + error))
+                        .build();
+
+                final Realm realm = Realm.getInstance(config);
+                looperThread.addTestRealm(realm);
+                assertFalse(Util.isEmptyString(config.getUser().getIdentity()));
```

Comment:
Probably also check that their identity starts with `anonymous_` or is that an implementation detail?

Reply:
I think it's impl details. user id I'm getting doesn't start with `anonymous_`